### PR TITLE
Added validation to make sure theme entries are colours

### PR DESCRIPTION
**Pull request by hmhealey**

### CODE REVIEW

File: webapp/utils/utils.jsx [hunk 1]
```diff
@@ -557,12 +557,16 @@ export function toTitleCase(str) {
 }
 
 export function applyTheme(theme) {
-    if (theme.sidebarBg) {
+    function isHexColor(value) {
+        return value && (/^#[0-9a-f]{6}$/i).test(value);
```
Comment (jwilander) [inline webapp/utils/utils.jsx]: Isn't it possible to have a hex colour that is just 3 in length? (ex. #fff)

Comment (hmhealey) [inline webapp/utils/utils.jsx]: Oh yeah, that's right. I'll add in a case for that

**Comment by crspeller** [comment]
Shouldn't we just validate the input instead of when we are applying it?

**Comment by hmhealey** [comment]
Good point, I moved it to the modal so it'll only be fired once when the user pastes the theme vector in

**Comment by crspeller** [comment]
Can we do server side validation as well? I know this stops any errors client side but server side is nice too.

**Comment by hmhealey** [comment]
Added serverside validation. It just blanks out invalid theme values as opposed to rejecting the user since doing so would break Mattermost for any user that already happens to have invalid theme values.

**Comment by crspeller** [comment]
+1

**Comment by jwilander** [comment]
+1

**Pull request merged**